src/main/java/com/snowflake/kafka/connector/internal/SnowflakeTelemetryBasicInfo.java

```diff
@@ -10,12 +17,87 @@
   static final String STAGE_NAME = "stage_name";
   static final String PIPE_NAME = "pipe_name";
 
+  // Name of mBean if jmx monitoring is enabled
+  private ObjectName mBeanName;
+
+  // A boolean to turn on or off a JMX metric as required.
+  private volatile boolean enableJMXMonitoring;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SnowflakeTelemetryBasicInfo.class);
+
+  private static final String JMX_METRIC_PREFIX = "snowflake.kafka.connector";
+
   SnowflakeTelemetryBasicInfo(
-      final String tableName, final String stageName, final String pipeName) {
+      final String tableName,
+      final String stageName,
+      final String pipeName,
+      final String connectorName,
+      final boolean enableJMXMonitoring) {
     this.tableName = tableName;
     this.stageName = stageName;
     this.pipeName = pipeName;
+    this.enableJMXMonitoring = enableJMXMonitoring;
```

Comment:
Nit: Any assert needed for connectorName?

Reply:
Name seems like a required property. So may be assert is not needed. (And we do throw exception if name is empty or invalid)
https://docs.snowflake.com/en/user-guide/kafka-connector-install.html#required-properties